src/Parallel/spanningtrees/boruvka-distributed.jl

```diff
@@ -0,0 +1,129 @@
+#=
+boruvka_multithread_lightgraph:
+- Julia version: 1.3.1
+- Author: hanstananda
+- Date: 2020-03-15
+=#
+@everywhere using LightGraphs, SimpleWeightedGraphs
+using CPUTime
+using DelimitedFiles
+@everywhere using Base
+@everywhere using DataStructures
+using SharedArrays
+
+"""
+    boruvka_mst_distributed(g)
+
+Return a tuple `(mst, weights)` where `mst` is a vector of edges representing the
+optimum (minimum, by default) spanning tree of a connected, undirected graph
+`g` , and `weights` is the sum of all the edges in the solution by using
+[Boruvka's algorithm](https://en.wikipedia.org/wiki/Bor%C5%AFvka%27s_algorithm).
+The algorithm requires that all edges have different weights to correctly generate a minimun/maximum spanning tree
+### Optional parameter(s):
+`maxItr`: Used to limit maximum number of iterations the algorithm will be performed. The default is log2(numVertex).
+"""
+function boruvka_mst_distributed(g::SimpleWeightedGraph, maxItr = convert(Int64, round(log2(nv(g))+1, digits=0)))
+    connected_vs = IntDisjointSets(nv(g))
+    joined_nodes = Dict{Int, Vector{Int}}(i=>[i] for i in 1:nv(g))
+    MAX_WEIGHT=2000000000.0
+    minCost = SharedArray{Float64}(nv(g)+1)
+    minNodeTgt = SharedArray{Int}(nv(g)+1)
+    minNodeSrc = SharedArray{Int}(nv(g)+1)
+    mst = Vector{edgetype(g)}()
+    sizehint!(mst, nv(g) - 1)
+    res =0
+    i=1
+    println("Max iteration: ", maxItr)
+    while(i<maxItr && length(mst)< nv(g)-1)
+        i+=1
+        initMinCostArray(g, minNodeSrc, minNodeTgt, minCost, MAX_WEIGHT)
+        findMinCostVertex(g, minNodeSrc, minNodeTgt, minCost, joined_nodes, connected_vs)
+        res+= contractVertex(g, minNodeSrc, minNodeTgt, minCost, joined_nodes, connected_vs,
+                mst, MAX_WEIGHT)
+    end
+
+    return (mst=mst, weight=res)
+end
+
+function initMinCostArray(
+        g::SimpleWeightedGraph,
+        minNodeSrc::SharedArray{Int,1},
+        minNodeTgt::SharedArray{Int,1},
+        minCost::SharedArray{Float64,1},
+        MAX_WEIGHT::Float64
+    )
+    for i = 1:nv(g)
```

Comment:
Think about whether you really want to iterate over the number of vertices, or the vertices themselves. That is,

`for i = 1:nv(g)` vs `for i in vertices(g)`.

Reply:
Hm after I think about it the latter one is much better.